lib/ansible/modules/cloud/amazon/aws_ses_identity_dkim.py

```diff
@@ -0,0 +1,223 @@
+            # other parts of the SES API. So if we get this specific client error we retry after
+            # a delay  so that we correctly enable or disable DKIM.
+            # Note we need this even when disabling since this seems to fail tests sometimes even
+            # after having succeeded once. It seems that we get inconsistent failures for some time
+            # until the domain verification attempt is fully propagated.
+            if (retry >= retries or
+                    not (error_code == 'InvalidParameterValue' and 'not verified for DKIM signing' in error_message)):
+                module.fail_json_aws(e, msg='Failed to {operation} DKIM for {identity}.'.format(
+                    operation=operation,
+                    identity=identity,
+                ))
+            time.sleep(retry_delay)
+
+
+def main():
+    module = AnsibleAWSModule(
+        argument_spec={
+            "identity": dict(required=True, type='str'),
+            "state": dict(default='enabled', choices=['enabled', 'disabled']),
+        },
+        supports_check_mode=True,
+    )
+
+    # SES APIs seem to have a much lower throttling threshold than most of the rest of the AWS APIs.
+    # Docs say 1 call per second. This shouldn't actually be a big problem for normal usage, but
+    # the ansible build runs multiple instances of the test in parallel that's caused throttling
+    # failures so apply a jittered backoff to call SES calls.
+    client = module.client('ses', retry_decorator=AWSRetry.jittered_backoff())
+
+    identity = module.params.get('identity')
+    state = module.params.get('state')
+    changed = False
+
+    state_check = _get_state_check(state)
+
+    # Get current DKIM attributes to see if they need changing
+    # Note that the get_identity_dkim_settings seems to take a long time to consistently return the
+    # updated result. So to make the tests reliable we force a wait here till we get to the expected
+    # state. Simply waiting for the correct return on change isn't sufficient since the old state seems
+    # to still be randomly returned for some time.
+    #
+    # This should only present a problem when rapidly switching back and forth as we do in tests, so
+    # this should not cause failures in real usage.
+    attributes = get_identity_dkim_settings(module, client, identity,
+                                            _get_state_check(os.environ.get('TEST_ONLY_FORCE_INITIAL_STATE', 'any')))
+    if not state_check(attributes):
+        # Update DKIM settings
+        if state == 'enabled':
+            ses_verify_dkim_domain(module, client, identity)
+        set_identity_dkim_enabled(module, client, identity, state == 'enabled')
+        changed = True
+        wait_for = state_check
+        if module.check_mode:
+            wait_for = _get_state_check('any')
+        attributes = get_identity_dkim_settings(module, client, identity, wait_for)
```

Comment:
This is where we should have the special handling of the attributes in check_mode rather than in get_identity_dkim attributes since we only want to skip the call after the change not the initial state check. So 215-217 should become:

    if module.check_mode:
        attributes = {
            'DkimEnabled': state == 'enabed'
        }
    else:
        attributes = get_identity_dkim_settings(module, client, identity, wait_for)

I.e.

* when not check mode, call get_identity_dkim_settings and wait for it to be in the correct state.
* when check mode, just create the expected attribute structure as if we made the change.